src/MetadataManager/PathToHash.ts

```diff
@@ -0,0 +1,269 @@
+    const baseUri =
+        vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, '.meta/hash_objects');
+    const files = await vscode.workspace.fs.readDirectory(baseUri);
+    for (const file of files) {
+      const hashFolderUri = vscode.Uri.joinPath(baseUri, file[0]);
+      const hashList = await vscode.workspace.fs.readDirectory(hashFolderUri);
+      for (const hashFile of hashList) {
+        const hashUri = vscode.Uri.joinPath(hashFolderUri, hashFile[0]);
+        let metaObj =
+            JSON.parse(Buffer.from(await vscode.workspace.fs.readFile(hashUri)).toString());
+        const hash = file[0] + hashFile[0].split('.')[0];
+        for (const path in metaObj) {
+          const uri = vscode.Uri.file(path);
+          // Metadata.disable(uri, hash);
+        }
+      }
+    }
+  }
+}
+
+export class PathToHash {
+  private static _instance: PathToHash;
+  private _map: any;
+
+  private constructor() {}
+
+  public static async getInstance() {
+    if (!this._instance) {
+      this._instance = new PathToHash();
+      this._instance._map = await this._instance.init();
+      await SyncMetadata.validateMetadata(await this._instance.getFlatMap());
+    }
+    return this._instance;
+  }
+
+  private async init() {
+    if (vscode.workspace.workspaceFolders === undefined) {
+      return;
+    }
+    const uri = vscode.workspace.workspaceFolders[0].uri;
+
+    return this.scanRecursively(uri);
+  }
+
+  private async scanRecursively(uri: vscode.Uri) {
+    let subMap: {[key: string]: any} = {};
+    const files = await vscode.workspace.fs.readDirectory(uri);
+
+    for (const file of files) {
+      const name: string = file[0];
+      const type: number = file[1];
+
+      if (type === 1) {
+        subMap[name] = await PathToHash.generateHash(vscode.Uri.joinPath(uri, '/' + name));
+      } else if (type === 2 && name !== '.meta') {
+        subMap[name] = await this.scanRecursively(vscode.Uri.joinPath(uri, '/' + name));
+      }
+    }
+    if(subMap === {}) {
+      return;
+    }
+
+    return subMap;
+  }
+
+  async getFlatMap() {
+    let flatMap: any = {};
+    let map = this._map;
+    let queue = [];
+    for (let data in map) {
+      queue.push([data, map[data], data.toString()]);
+    }
+
+    while (queue.length !== 0) {
+      const subMap = queue.pop();
+      if (subMap === undefined) {
+        continue;
+      }
+      if (vscode.workspace.workspaceFolders === undefined) {
+        break;
+      }
+      let path =
+          vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, subMap[2]).path;
+      if (fs.lstatSync(path).isDirectory()) {
+        for (let key in subMap[1]) {
+          queue.push([key, subMap[1][key], subMap[2] + '/' + key]);
+        }
+      } else {
+        flatMap[subMap[2]] = subMap[1];
+      }
+    }
+    return flatMap;
+  }
+
+
+  /**
+   * 
+   * @param uri 
+   * @returns hash :string
+   * @brief put uri in the '_map' to get a hash
+   */
+  get(uri: vscode.Uri) {
```

Comment:
Could you rename the function by specifying what it 'get'..?

Reply:
What dose it look like to rename this function to 'getHash'?

Reply:
Sure :-D